### PR TITLE
Migrate to `humantime`

### LINKED ISSUE #2085: chore: Migrate to `humantime` config parser
### Background
Currently, units of config params with the `Duration` type are hard coded and being parsed using custom functions like `duration_from_seconds`. In order to switch between the units it's required to change the code which is not flexible. 

### Details
The humantime library allows us to specify durations in a more human-readable format (e.g., 1s, 2m, 3h). Currently, our configurations use plain numbers (interpreted as seconds), which can be less intuitive and error-prone. This migration will be executed in two distinct phases to ensure smooth transition and maintain compatibility with our existing deployment processes, specifically with Pulumi configurations.

- [ ] Phase 1: Supporting Both Formats: The primary goal of the first phase is to update the application to support both the existing plain number format and the new humantime format simultaneously. This is crucial for maintaining system stability and ensuring uninterrupted operations during the migration process.
- [ ] Phase 2: Completing the Migration to humantime: Once all the configurations in the Pulumi repository have been updated to the new humantime format, we will initiate the second phase.

### Acceptance criteria

- [ ] All the config params with Duration-intended types use `humantime` parser.

### PR DESCRIPTION
**Pull request by squadgazzz**
# Description
Fully closes the https://github.com/cowprotocol/services/issues/2085. Since DEX solvers are about to migrate to a separate repo, it's a good time to change the config format. It will be much more difficult later.

# Changes
Use `humantime` to parse `Duration` config params.

## How to test
e2e + staging

## Related Issues

Fixes #2085

### CODE REVIEW

**Review by fleupold — APPROVED** [review]
Great initiative and great change!

File: crates/e2e/src/setup/colocation.rs [hunk 1]
```diff
@@ -78,7 +78,7 @@ graph-api-base-url = "https://api.thegraph.com/subgraphs/name/"
 [[liquidity.uniswap-v2]]
 router = "{:?}"
 pool-code = "{:?}"
-missing-pool-cache-time-seconds = 3600
+missing-pool-cache-time = "10m"
```
Comment (MartinquaXD) [inline crates/e2e/src/setup/colocation.rs:81]: nit: The default changed from `1h` to `10m`.

File: crates/solver/src/arguments.rs [hunk 2]
```diff
@@ -415,20 +415,16 @@ impl std::fmt::Display for Arguments {
             "max_additional_eden_tip: {}",
             self.max_additional_eden_tip
         )?;
-        writeln!(
-            f,
-            "max_submission_seconds: {:?}",
-            self.max_submission_seconds
-        )?;
+        writeln!(f, "max_submission: {:?}", self.max_submission)?;
```
Comment (MartinquaXD) [inline crates/solver/src/arguments.rs:418]: nit: Now it's no longer obvious that this is a time period. Maybe `max_submission_time`?

**Review by MartinquaXD — APPROVED** [review]
Very nice!

File: crates/e2e/src/setup/colocation.rs (hunk 1 above)
Comment (squadgazzz) [reply crates/e2e/src/setup/colocation.rs:81]: Overlooked, thanks

**Pull request merged**